web-ui/src/pages/FeedbackRequestPage.jsx

```diff
@@ -72,60 +76,98 @@ function getTemplates() {
   ];
 }
 
-let todayDate = new Date()
 const FeedbackRequestPage = () => {
   const steps = getSteps();
   const classes = useStyles();
   const location = useLocation();
   const history = useHistory();
-
   const query = queryString.parse(location?.search);
 
   const stepQuery = query.step?.toString();
+  const templateQuery = query.template?.toString();
+  const fromQuery = query.from?.toString();
+  const dueQuery = query.due?.toString();
+
+  const getStep = useCallback(() => {
+    if(!stepQuery || stepQuery < 1 || !/^\d+$/.test(stepQuery))
+      return 1;
+    else return parseInt(stepQuery);
+  },[stepQuery]);
 
-  let sendDate = query?.sendDate ? query.sendDate: todayDate.toString();
-  let dueDate = query?.dueDate ? query.dueDate: null
-  let activeStep = location?.search ? parseInt(stepQuery) : 1;
-  const numbersOnly = /^\d+$/.test(stepQuery);
+  const activeStep = getStep();
 
   const [preview, setPreview] = useState({open: false, selectedTemplate: null});
 
   const handlePreviewOpen = (event, selectedTemplate) => {
+    event.stopPropagation();
     setPreview({open: true, selectedTemplate: selectedTemplate});
-  }
+  };
 
   const handlePreviewClose = (selectedTemplate) => {
     setPreview({open: false, selectedTemplate: selectedTemplate});
-  }
+  };
 
-  const getFeedbackArgs = (step) => {
-    const nextQuery = {
-      ...query,
-      step: step
-    }
-
-    return `/feedback/request/?${queryString.stringify(nextQuery)}`;
+  const onCardClick = (template) => {
+    history.push(`/feedback/request/?template=${template.id}`);
   }
 
-  const handleQueryChange = (key, value) => {
-    let newQuery = {
-      ...query,
-      [key]: value
+  const hasTemplate = useCallback(() => {
+    return !!templateQuery;
+  }, [templateQuery])
+
+  const hasFrom = useCallback(() => {
+    return !!fromQuery;
+  }, [fromQuery])
+
+  const isValidDate = useCallback((dateString) => {
+    const timestamp = Date.parse(dateString);
+    return !isNaN(timestamp);
+  }, []);
+
+  const hasDue = useCallback(() => {
+    return (dueQuery && isValidDate(dueQuery))
+  }, [dueQuery, isValidDate]);
```

Comment:
There is no validation here to make sure that the send date and the due date are on or later than the current date. We could potentially have a separate story for this since it would involve displaying error messages for the text fields.

Reply:
Definitely would split into another story. There probably needs to be some discussion about whether validation like that is a server or UI concern too.

Reply:
Split into a separate issue at #1098